llvm/lib/Transforms/Scalar/JumpThreading.cpp

```diff
@@ -1269,6 +1269,7 @@ bool JumpThreadingPass::simplifyPartiallyRedundantLoad(LoadInst *LoadI) {
     if (IsLoadCSE) {
       LoadInst *NLoadI = cast<LoadInst>(AvailableVal);
       combineMetadataForCSE(NLoadI, LoadI, false);
+      LVI->eraseBlock(NLoadI->getParent());
```

Comment:
It's not necessary to erase the whole block here. Just eraseValue() on the load should be enough, though you may need to publicly expose the API first (it's usually automatically called via ValueHandle).

Reply:
I exposed the API. I'm learning how to get ValueHandle to handle this automatically. This may take some time.